src/torchmetrics/nominal/tschuprows.py

```diff
@@ -0,0 +1,102 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, Optional, Union
+
+import torch
+from torch import Tensor
+from typing_extensions import Literal
+
+from torchmetrics.functional.nominal.tschuprows import _tschuprows_t_compute, _tschuprows_t_update
+from torchmetrics.functional.nominal.utils import _nominal_input_validation
+from torchmetrics.metric import Metric
+
+
+class TschuprowsT(Metric):
+    r"""Compute `Tschuprow's T`_ statistic measuring the association between two categorical (nominal) data series.
+
+    .. math::
+        T = \sqrt{\frac{\chi^2 / n}{\sqrt{(r - 1) * (k - 1)}}}
+
+    where
+
+    .. math::
+        \chi^2 = \sum_{i,j} \ frac{\left(n_{ij} - \frac{n_{i.} n_{.j}}{n}\right)^2}{\frac{n_{i.} n_{.j}}{n}}
+
+    Tschuprow's T is a symmetric coefficient, i.e.
+
+    .. math::
+        T(preds, target) = T(target, preds)
```

Comment:
nitpicking: I think we should make this inline math instead for all the metrics
```suggestion
    Tschuprow's T is a symmetric coefficient, i.e. :math:`T(preds, target) = T(target, preds)`.
```